### PR TITLE
feat: add message start event support

### PR DESCRIPTION
**Pull request by igpetrov**
feat: add message start event support

Note: other connectors will be supplied later after I complete with testing at least Kafka, and SQS.

### CODE REVIEW

File: connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java [hunk 1]
```diff
@@ -103,6 +109,52 @@ protected InboundConnectorResult<ProcessInstance> triggerStartEvent(
...
+      PublishMessageResponse result =
+          zeebeClient
+              .newPublishMessageCommand()
+              .messageName(correlationPoint.messageName())
+              .correlationKey("") // must be empty to start new process
+              // https://docs.camunda.io/docs/components/modeler/bpmn/message-events/#message-start-events
+              .tenantId(definition.tenantId())
+              .variables(extractedVariables)
+              .send()
+              .join();
```
Comment (sbuettner) [inline connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java:139]: I think we should support the `messageId` here.

File: connector-sdk/core/src/main/java/io/camunda/connector/api/inbound/correlation/MessageStartEventCorrelationPoint.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.api.inbound.correlation;
+
+public record MessageStartEventCorrelationPoint(
+    String messageName, String bpmnProcessId, int version, long processDefinitionKey)
+    implements ProcessCorrelationPoint {
```
Comment (sbuettner) [inline connector-sdk/core/src/main/java/io/camunda/connector/api/inbound/correlation/MessageStartEventCorrelationPoint.java:21]: We need to support messageId here.

File: connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java (hunk 1 above)
Comment (igpetrov) [reply connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java:139]: Good point. I thought that `id` is something internal to Zeebe, apparently we can set it ourselves. I will add it and test.

Comment (sbuettner) [reply connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java:139]: Yeah, its actually quite important and we need to support it also via the element templates (FEEL expression) as its the only way to achieve idempotency with zeebe.

Comment (igpetrov) [reply connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java]: Updated

File: connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java [hunk 3]
```diff
@@ -103,6 +110,70 @@ protected InboundConnectorResult<ProcessInstance> triggerStartEvent(
...
+          new CorrelationErrorData(CorrelationErrorReason.ACTIVATION_CONDITION_NOT_MET));
+    }
+
+    String idempotencyKeyMessageId = extractIdempotencyKey(correlationPoint, variables);
+    if (correlationPoint.idempotencyKeyExpression() != null
+        && !correlationPoint.idempotencyKeyExpression().isBlank()
+        && idempotencyKeyMessageId == null) {
+      LOG.info(
+          "Wasn't able to obtain idempotency key for expression {}.",
+          correlationPoint.idempotencyKeyExpression());
```
Comment (sbuettner) [inline connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java:131]: Could be `debug`.

File: connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java [hunk 4]
```diff
@@ -103,6 +110,70 @@ protected InboundConnectorResult<ProcessInstance> triggerStartEvent(
...
+      LOG.debug("Activation condition didn't match: {}", correlationPoint);
+      return new MessageStartCorrelationResult(
+          correlationPoint.messageName(),
+          new CorrelationErrorData(CorrelationErrorReason.ACTIVATION_CONDITION_NOT_MET));
+    }
+
+    String idempotencyKeyMessageId = extractIdempotencyKey(correlationPoint, variables);
+    if (correlationPoint.idempotencyKeyExpression() != null
+        && !correlationPoint.idempotencyKeyExpression().isBlank()
+        && idempotencyKeyMessageId == null) {
```
Comment (sbuettner) [inline connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java:128]: Shouldnt `idempotencyKeyMessageId != null` check be enough here? I assume that `extractIdempotencyKey` already checks it?

File: connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/importer/ProcessDefinitionInspector.java [hunk 5]
```diff
@@ -226,7 +228,26 @@ private Optional<ProcessCorrelationPoint> getCorrelationPointCatchEvent(CatchEve
...
+    MessageEventDefinition msgDef =
+        (MessageEventDefinition)
+            startEvent.getEventDefinitions().stream()
+                .filter(def -> def instanceof MessageEventDefinition)
+                .findAny()
+                .orElse(null);
+
+    if (msgDef != null) {
+      String idempotencyKey =
+          extractRequiredProperty(startEvent, IDEMPOTENCY_KEY_EXPRESSION_START_MESSAGE_EVENT);
```
Comment (sbuettner) [inline connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/importer/ProcessDefinitionInspector.java:242]:
```suggestion
      String messageIdExpression =
          extractRequiredProperty(startEvent, MESSAGE_ID_EXPRESSION_START_MESSAGE_EVENT);
```

File: connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/importer/ProcessDefinitionInspector.java [hunk 6]
```diff
@@ -226,7 +228,26 @@ private Optional<ProcessCorrelationPoint> getCorrelationPointCatchEvent(CatchEve
...
+      return Optional.of(
+          new MessageStartEventCorrelationPoint(
+              msgDef.getMessage().getName(),
+              idempotencyKey,
```
Comment (sbuettner) [inline connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/importer/ProcessDefinitionInspector.java:246]:
```suggestion
              messageId,
```

File: connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java (hunk 4 above)
Comment (igpetrov) [reply connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java:128]: This is more like a safety check: when expression is defined but we won't able to extract it. Without this check we will return `500`.

Comment (igpetrov) [reply connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java:128]: on its own, message id expression can be blank

File: connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/Keywords.java [hunk 7]
```diff
@@ -60,6 +60,9 @@ public class Keywords {
    */
   public static final String CORRELATION_KEY_EXPRESSION_KEYWORD = "correlationKeyExpression";
 
+  public static final String IDEMPOTENCY_KEY_EXPRESSION_START_MESSAGE_EVENT =
+      "idempotencyKeyExpression";
```
[the remark this answers is not in the document]
Comment (igpetrov) [reply connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/Keywords.java]: Done

File: connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java (hunk 3 above)
Comment (igpetrov) [reply connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java]: Done

File: connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/importer/ProcessDefinitionInspector.java (hunk 5 above)
Comment (igpetrov) [reply connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/importer/ProcessDefinitionInspector.java]: Done

File: connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/importer/ProcessDefinitionInspector.java (hunk 6 above)
Comment (igpetrov) [reply connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/importer/ProcessDefinitionInspector.java]: Done

**Review by sbuettner — APPROVED** [review]

**Pull request merged**